tools/shared.py

```diff
@@ -1192,6 +1192,10 @@ def __getattr__(self, attr):
         raise AttributeError("Settings object has no attribute '%s'" % attr)
 
     def __setattr__(self, attr, value):
+      if attr == 'STRICT' and value:
+        for a in self.legacy_settings:
+          self.attrs.pop(a, None)
```

Comment:
I don't understand this line - is it doing `pop` on a dictionary?

Reply:
Yes, basically its a way to remove an item from a dict regardless of whether it exists or not.  Should I add a comment?

Reply:
doesn't `del` do that?

Reply:
I think `del` will throw on missing keys.

Reply:
I see, sounds good. lgtm.

Btw, I don't have a full understanding of long-term plans for STRICT. I know @juj planned to eventually make it the default - is that still the goal?

Reply:
I have an issue open discussing that: https://github.com/emscripten-core/emscripten/issues/8317.

My understanding is that all the things that fall under STRICT today should eventually be the default, but that we can also continually add new things to the set.  So STRICT mode is an every changing set of things we hope to switch to one day.